sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/CreateTableHelpers.java

```diff
@@ -46,8 +48,28 @@ public class CreateTableHelpers {
    */
   private static Set<String> createdTables = Collections.newSetFromMap(new ConcurrentHashMap<>());
 
+  // When CREATE_IF_NEEDED is specified, BQ tables should be created if they do not exist. This
+  // method detects
+  // errors on table operations, and attempts to create the table if necessary.
+  static void createTableWrapper(Callable<Void> action, Callable<Boolean> tryCreateTable)
+      throws Exception {
+    for (int retry = 0; retry <= 1; ++retry) {
+      try {
+        action.call();
+      } catch (ApiException | StatusRuntimeException e) {
+        // TODO: Once BigQuery reliably returns a consistent error on table not found, we should
```

Comment:
Maybe an issues can be created in the client to track the feature request / improvement.